apps/cowswap-frontend/src/modules/permit/utils/handlePermit.ts

```diff
@@ -0,0 +1,36 @@
+import { AppDataInfo, buildAppDataHooks, updateHooksOnAppData } from 'modules/appData'
+import { generatePermitHook, HandlePermitParams } from 'modules/permit'
+
+/**
+ * Handle token permit
+ *
+ * Will request user signature if needed
+ * Can use cached permit if available
+ *
+ * If not needed, will remove any permit info from appData
+ *
+ * Returns the updated appData
+ */
+export async function handlePermit(params: HandlePermitParams): Promise<AppDataInfo> {
+  const { permitInfo, hasEnoughAllowance, inputToken, provider, account, chainId, appData } = params
+
+  if (permitInfo && !hasEnoughAllowance) {
+    // If token is permittable and there's not enough allowance, get the permit hook
+
+    // TODO: maybe we need a modal to inform the user what they need to sign?
```

Comment:
Are we keeping track of potential reiterations? 
Isn't this connected to the modals @fairlighteth made?

Reply:
Will create issues for the TODOs
Should use Michel's modals, yes!

Reply:
https://github.com/cowprotocol/cowswap/issues/3145

Reply:
Thanks